### PR TITLE
refactor(esl-core): fix types and documentation

### PR DESCRIPTION
**Pull request by ala-n**

### CODE REVIEW

File: src/modules/esl-base-element/README.md [hunk 1]
```diff
@@ -1,28 +1,49 @@
...
 
-### Example
+### Base Element API
+Properties:
+- `connected` - readonly marker, true if element connected and base `connectedCallback` executed
+
+Attributes: 
+- `connectedCallback` - called when element appends to the DOM
+- `disconnectedCallback` - called when element is disconnected from the DOM
+- `attributeChangeCallback` - called when observed attribute has changed
```
Comment (abarmina) [inline src/modules/esl-base-element/README.md:28]:
```suggestion
- `attributeChangeCallback` - called when the observable attribute is changed
```

File: src/modules/esl-base-element/README.md [hunk 2]
```diff
@@ -1,28 +1,49 @@
...
+
+- `$$cls` - check or change element CSS classes (uses CSSClassUtils)
```
Comment (abarmina) [inline src/modules/esl-base-element/README.md:30]:
```suggestion
- `$$cls` - check or change element CSS classes (using CSSClassUtils) 
```

File: src/modules/esl-base-element/core/esl-base-element.ts [hunk 3]
```diff
@@ -25,22 +25,33 @@ export abstract class ESLBaseElement extends HTMLElement {
...
     return value;
   }
 
-  /** Get or set attribute */
+  /**
+   * Gets or sets attribute for current element.
+   * If the `value` param is undefined then skips mutation.
+   * @param name - attribute name
+   * @param value - string attribute value, boolean attribute state or `null` to delete attribute
+   * @returns current attribute value or previous value for mutation
```
Comment (abarmina) [inline src/modules/esl-base-element/core/esl-base-element.ts:45]:
```suggestion
   * @returns the current attribute value or previous value to mutate
```

**Review by abarmina — APPROVED** [review]

File: src/modules/esl-base-element/README.md (hunk 2 above)
Comment (ala-n) [reply src/modules/esl-base-element/README.md:30]: It looks like using will have a different meaning here

File: src/modules/esl-base-element/core/esl-base-element.ts [hunk 4]
```diff
@@ -25,22 +25,33 @@ export abstract class ESLBaseElement extends HTMLElement {
...
+   */
   public $$cls(cls: string, value?: boolean): boolean {
     if (value === undefined) return CSSClassUtils.has(this, cls);
     CSSClassUtils.toggle(this, cls, value);
     return value;
   }
 
-  /** Get or set attribute */
+  /**
+   * Gets or sets attribute for current element.
```
[the remark this answers is not in the document]
Comment (ala-n) [reply src/modules/esl-base-element/core/esl-base-element.ts:41]: Fixed the missing article
The method can work with the single attribute at the time

Comment (ala-n) [inline src/modules/esl-base-element/core/esl-base-element.ts:41]:
```suggestion
   * Gets or sets an attribute for the current element.
```

File: src/modules/esl-base-element/README.md (hunk 2 above)
Comment (abarmina) [reply src/modules/esl-base-element/README.md:30]: yes, you are right

**Review by yadamskaya — APPROVED** [review]

**Pull request merged**